src/misc.ts

```diff
@@ -32,10 +32,14 @@ export type PartialOrAbsent<Value> = Partial<Value> | null | undefined;
 export type NonEmptyArray<Element> = [Element, ...Element[]];
 
 /**
- * A JavaScript object that is not `null`, a function, or an array. The object
- * can still be an instance of a class.
+ * Any type that can be used as the name of an object property.
  */
-export type RuntimeObject = Record<number | string | symbol, unknown>;
+export type ValidPropertyType = string | number | symbol;
```

Comment:
Perhaps this could be named something like `PropertyKey`?

Reply:
Good idea, I wasn't happy with this name either

Reply:
Oh, apparently this is already a global baked-into TypeScript! I have replaced this custom type and switched to using the built-in one instead.